src/middlewared/middlewared/plugins/unscheduled_reboot_alert.py

```diff
@@ -0,0 +1,133 @@
+from datetime import datetime
+import os
+import textwrap
+import re
+
+from middlewared.service import Service
+from middlewared.client import Client
+
+SENTINEL_PATH = "/data/sentinels/unscheduled-reboot"
+MIDDLEWARE_STARTED_SENTINEL_PATH = "/tmp/.middleware-started"
+
+# This file is managed in TrueNAS HA code (carp-state-change-hook.py)
+# Ticket 39114
+WATCHDOG_ALERT_FILE = "/data/sentinels/.watchdog-alert"
+
+# This file is managed in TrueNAS HA code (.../sbin/fenced)
+# Ticket 39114
+FENCED_ALERT_FILE = "/data/sentinels/.fenced-alert"
+
+# This file only exists on TrueNAS HA systems
+# Ticket 39114
+TRUENAS_HA_FILE = "/tmp/.ha_mode"
+
+
+class UnscheduledRebootAlertService(Service):
+    async def terminate(self):
+        if os.path.exists(SENTINEL_PATH):
+            os.unlink(SENTINEL_PATH)
+
+
+async def setup(middleware):
+    if os.path.exists(SENTINEL_PATH):
+        # We want to emit the mail only if the machine truly rebooted
+        if os.path.exists(MIDDLEWARE_STARTED_SENTINEL_PATH):
+            return
+
+        gc = await middleware.call('datastore.config', 'network.globalconfiguration')
+        now = datetime.now().strftime("%c")
+
+        Client() = c
+        if not c.call('notifier.is_freenas') and c.call('notifier.failover_licensed')
+            try:
+                with open(TRUENAS_HA_FILE, "rb") as controller:
+                    match = re.search(r"(.*):(.*)")
+                if match.group(2) == 'B':
+                    hostname = f"{gc['gc_hostname_b'].{gc['gc_domain']}"
```

Comment:
You're not handling node = 'A"

Also, better use: notifier.failover_node call to get B or A